src/soilbiogeochem/SoilBiogeochemNitrogenStateType.F90

```diff
@@ -329,22 +404,57 @@ subroutine InitCold(this, bounds, &
           do j = 1, nlevdecomp
              do k = 1, ndecomp_pools
                 this%decomp_npools_vr_col(c,j,k) = decomp_cpools_vr_col(c,j,k) / decomp_cascade_con%initial_cn_ratio(k)
-             end do
+                if(use_soil_matrixcn)then
+                   this%matrix_cap_decomp_npools_vr_col(c,j,k) = decomp_cpools_vr_col(c,j,k) / decomp_cascade_con%initial_cn_ratio(k)
+                   this%in_nacc_2d(c,j,k) = 0._r8
+                   this%vert_up_tran_nacc(c,j,k) = 0._r8
+                   this%vert_down_tran_nacc(c,j,k) = 0._r8
+                   this%exit_nacc(c,j,k) = 0._r8
+                   this%decomp0_npools_vr_col(c,j,k) = amax1(this%decomp_npools_vr_col(c,j,k),1.e-30_r8)
+                end if
+            end do
+            if(use_soil_matrixcn)then
+               do k = 1, ndecomp_cascade_transitions
+                  this%hori_tran_nacc(c,j,k) = 0._r8
+               end do
+            end if
+
              this%sminn_vr_col(c,j) = 0._r8
              this%ntrunc_vr_col(c,j) = 0._r8
           end do
+
```

Comment:
This probably should have a "if(use_soil_matrixcn)" if statement around it

Reply:
done